integrations/aws-v3/aws/auth/session_factory.py

```diff
@@ -7,42 +7,78 @@
 from loguru import logger
 from port_ocean.context.ocean import ocean
 from aiobotocore.session import AioSession
-from typing import Any, TypedDict, AsyncIterator
+from typing import Any, TypedDict, AsyncIterator, Dict
 from aws.auth.providers.assume_role_with_web_identity_provider import (
     AssumeRoleWithWebIdentityProvider,
 )
 import os
 
 StrategyType = SingleAccountStrategy | MultiAccountStrategy | OrganizationsStrategy
 
+DEFAULT_PROVIDER_PRIORITY: str = "AssumeRoleWithWebIdentity,StaticCredential,AssumeRole"
+
 
 class AccountStrategyFactory:
     """A factory for creating account strategies based on the global configuration."""
 
     _cached_strategy: StrategyType | None = None
 
+    _provider_factories: Dict[str, type[CredentialProvider]] = {
+        "AssumeRoleWithWebIdentity": AssumeRoleWithWebIdentityProvider,
+        "StaticCredential": StaticCredentialProvider,
+        "AssumeRole": AssumeRoleProvider,
+    }
+
     @classmethod
-    def _detect_provider_type(cls, config: dict[str, Any]) -> CredentialProvider:
+    def _get_provider_priority(cls, config: dict[str, Any]) -> list[str]:
         """
-        Detect the appropriate provider type based on environment variables and config.
-        Returns a tuple of (provider_instance, provider_type_name, strategy_class)
+        Returns the credential provider priority list, falling back to DEFAULT_PROVIDER_PRIORITY if not provided.
+        Example: "AssumeRoleWithWebIdentity,StaticCredential,AssumeRole"
         """
+        priority_str = (
+            config.get("credential_provider_priority") or DEFAULT_PROVIDER_PRIORITY
+        )
+        return [
+            provider.strip() for provider in priority_str.split(",") if provider.strip()
+        ]
 
-        # Check for web identity token first (highest priority)
-        if os.getenv("AWS_WEB_IDENTITY_TOKEN_FILE"):
-            logger.info(
-                "[AccountStrategyFactory] Using AssumeRoleWithWebIdentityProvider (found AWS_WEB_IDENTITY_TOKEN_FILE)"
-            )
-            return AssumeRoleWithWebIdentityProvider(config=config)
+    @classmethod
+    def _detect_provider_type(cls, config: dict[str, Any]) -> CredentialProvider:
+        """
+        Dynamically detect the appropriate provider type based on the configured priority.
+        """
+        for provider_name in cls._get_provider_priority(config):
+            provider_cls = cls._provider_factories.get(provider_name)
+            if not provider_cls:
+                logger.warning(
+                    f"[AccountStrategyFactory] Unknown provider '{provider_name}', skipping..."
+                )
+                continue
+
+            if cls._provider_is_applicable(provider_name, config):
+                logger.info(
+                    f"[AccountStrategyFactory] Using {provider_cls.__name__} (priority: {provider_name})"
```

Comment:
```suggestion
                    f"Using {provider_cls.__name__} (priority: {provider_name})"
```
Let's make sure that this is also user readable